pkg/cmd/codespace/create_test.go

```diff
@@ -382,17 +426,114 @@ Alternatively, you can run "create" with the "--default-permissions" option to c
+				useWeb: true,
+			},
+			wantURL: "https://github.com/codespaces/new",
+		},
+		{
+			name: "skip machine check when using web flag and no machine provided",
+			fields: fields{
+				apiClient: apiCreateDefaults(&apiClientMock{
+					GetRepositoryFunc: func(ctx context.Context, nwo string) (*api.Repository, error) {
+						return &api.Repository{
+							ID:            123,
+							DefaultBranch: "main",
+						}, nil
+					},
+					CreateCodespaceFunc: func(ctx context.Context, params *api.CreateCodespaceParams) (*api.Codespace, error) {
+						return &api.Codespace{
+							Name: "monalisa-dotfiles-abcd1234",
+						}, nil
+					},
+				}),
+			},
+			opts: createOptions{
+				repo:     "monalisa/dotfiles",
+				useWeb:   true,
+				branch:   "custom",
+				location: "EastUS",
+			},
+			wantStderr: "  ✓ Codespaces usage for this repository is paid for by monalisa\n",
+			wantURL:    fmt.Sprintf("https://github.com/codespaces/new?repo=%d&ref=%s&machine=%s&location=%s", 123, "custom", "", "EastUS"),
+		},
+		{
+			name: "return correct url with correct params when using web flag and repo flag",
+			fields: fields{
+				apiClient: apiCreateDefaults(&apiClientMock{
+					GetRepositoryFunc: func(ctx context.Context, nwo string) (*api.Repository, error) {
+						return &api.Repository{
+							ID:            123,
+							DefaultBranch: "main",
+						}, nil
+					},
+					CreateCodespaceFunc: func(ctx context.Context, params *api.CreateCodespaceParams) (*api.Codespace, error) {
+						return &api.Codespace{
+							Name: "monalisa-dotfiles-abcd1234",
+						}, nil
+					},
+				}),
+			},
+			opts: createOptions{
+				repo:   "monalisa/dotfiles",
+				useWeb: true,
+			},
+			wantStderr: "  ✓ Codespaces usage for this repository is paid for by monalisa\n",
+			wantURL:    fmt.Sprintf("https://github.com/codespaces/new?repo=%d&ref=%s&machine=%s&location=%s", 123, "main", "", ""),
+		},
+		{
+			name: "return correct url with correct params when using web flag, repo, branch, location, machine flag",
+			fields: fields{
+				apiClient: apiCreateDefaults(&apiClientMock{
+					GetRepositoryFunc: func(ctx context.Context, nwo string) (*api.Repository, error) {
+						return &api.Repository{
+							ID:            123,
+							DefaultBranch: "main",
+						}, nil
+					},
+					CreateCodespaceFunc: func(ctx context.Context, params *api.CreateCodespaceParams) (*api.Codespace, error) {
+						return &api.Codespace{
+							Name:    "monalisa-dotfiles-abcd1234",
+							Machine: api.CodespaceMachine{Name: "GIGA"},
+						}, nil
+					},
+				}),
+			},
+			opts: createOptions{
+				repo:     "monalisa/dotfiles",
+				machine:  "GIGA",
+				branch:   "custom",
+				location: "EastUS",
+				useWeb:   true,
+			},
+			wantStderr: "  ✓ Codespaces usage for this repository is paid for by monalisa\n",
+			wantURL:    fmt.Sprintf("https://github.com/codespaces/new?repo=%d&ref=%s&machine=%s&location=%s", 123, "custom", "GIGA", "EastUS"),
+		},
```

Comment:
Can we have a test for a wrong machine type?

Reply:
Okay,
added in [f55fd79](https://github.com/cli/cli/pull/7288/commits/f55fd79142632e36cbd28e4f3add2fd2c3f898a7)